### PR TITLE
Add new blog post about the signer API

### PR DESCRIPTION
**Pull request by jku**
This gives some example application code for the new securesystemslib.signer API.

### CODE REVIEW

**Comment by jku** [comment]
I think the easiest way to review is to read https://github.com/jku/python-tuf/blob/securesystemslib-blog/docs/_posts/2023-01-24-securesystemslib-signer-api.md -- only the header block is unhandled, otherwise it _should_ look like the published post

**Comment by jku** [comment]
CC @VickyMerzOwn: you mentioned you might be interested in reviewing

File: docs/_posts/2023-01-24-securesystemslib-signer-api.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+1. Store the public key _and the information needed to access the private key_
+1. Sign using the information stored in step 2
+1. Verify signature with public key
+
+Securesystemslib 0.26 introduces an improved signer API that recognizes this process complexity – and in turn makes managing and signing with keys simpler in practical application development. There are three main changes, all in the `securesystemslib.signer` module that defines Signer and Key classes:
+* The concept of **Private key URIs** is introduced – this is a relatively simple string that identifies a signing technology and encodes how to access and sign with a specific private key. Examples:
+  - `gcpkms:projects/python-tuf-kms/locations/global/keyRings/git-repo-demo/cryptoKeys/online/cryptoKeyVersions/1` (A Google Cloud KMS key)
+  - `file:/home/jku/keys/mykey?encrypted=true` (A key in an encrypted file)
+  - `hsm:` (A hardware security module like Yubikey)
+* **Importing** private key URIs and public keys is handled by Signers (there's no generic API though: this detail is specific to signing technology)
```
Comment (lukpueh) [inline docs/_posts/2023-01-24-securesystemslib-signer-api.md]: Minor nit: I have a bit of a problem with the wording "import private key URI". It sounds like the URI already exists somewhere and we move it from one place to another. In my mind, the URI is created during public key import.

File: docs/_posts/2023-01-24-securesystemslib-signer-api.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+* **Dynamic dispatch** is added for both Signers and Keys (former based on the private key URI, latter on the key content): As a result application code does not need to care about the specific technology used to sign/verify but securesystemslib can still support a wide array of signing methods -- and this support can even be extended with out-of-tree implementations.
+
+## Code examples
+
+These examples are slightly simplified copies from my latest repository implementation and should represent any new application code using the python-tuf Metadata API in the future[^1]. Some things to note in these examples:
+* Application code that signs does not care what signing technology is used
+* Import is specific to the underlying technology
```
Comment (lukpueh) [inline docs/_posts/2023-01-24-securesystemslib-signer-api.md]: Maybe...?
```suggestion
* Public key import and private key URI assembly is specific to the underlying technology
```
Just to make it clear that this is different from importing private keys, which was part of the old API.

**Review by lukpueh — APPROVED** [review]
Super cool post! I left two suggestions inline (feel free to ignore). Did you consider adding a third code example that showcases dynamic dispatch for public key and verification?

File: docs/_posts/2023-01-24-securesystemslib-signer-api.md (hunk 1 above)
Comment (jku) [reply docs/_posts/2023-01-24-securesystemslib-signer-api.md]: you're not wrong

**Comment by jku** [comment]
Amended the commit with the two minor suggestions: both clarify that private key URIs are _constructed_ while public keys are _imported_ from the underlying signing system.

**Comment by jku** [comment]
> Did you consider adding a third code example that showcases dynamic dispatch for public key and verification?

I did... Since the application code looks exactly what it looked like before, it's not that interesting.

We of course know that the dispatch will enable GPG or sigstore keys but that's not  really visible right now

**Pull request merged**